jdbc-v2/src/test/java/com/clickhouse/jdbc/PreparedStatementTest.java

```diff
@@ -222,4 +225,33 @@ public void testBigDecimal() throws Exception {
             }
         }
     }
+
+    @Test(groups = { "integration" })
+    public void testPrimitiveArrays() throws Exception {
+        try (Connection conn = getJdbcConnection()) {
+            try (PreparedStatement stmt = conn.prepareStatement("SELECT ?")) {
+                stmt.setObject(1, new String[][] {new String[]{"a"}, new String[]{"b"}, new String[]{"c"}});
+                try (ResultSet rs = stmt.executeQuery()) {
+                    assertTrue(rs.next());
+                    assertEquals(((BinaryStreamReader.ArrayValue)((BinaryStreamReader.ArrayValue)rs.getObject(1)).get(0)).get(0), "a");
+                    assertFalse(rs.next());
+                }
+            }
+        }
+    }
+
+
+    @Test(groups = { "integration" })
+    public void testEscapeStrings() throws Exception {
+        try (Connection conn = getJdbcConnection()) {
+            try (PreparedStatement stmt = conn.prepareStatement("SELECT FALSE OR ? = 'test'")) {
+                stmt.setString(1, "test\\\\' OR 1 = 1 --");
+                try (ResultSet rs = stmt.executeQuery()) {
+                    assertTrue(rs.next());
+                    assertEquals(rs.getString(1), "false");
```

Comment:
I think we need positive case, too to make sure the string reached the server in expected form